### PR TITLE
Adding ability to include attachment in system-err

### PR DESCRIPTION
**Pull request by anto-wahanda**
This is to implement the ability to add attachments (limited to system-err for now) as explained in the link below, so that they can be picked up by the jenkins plugin.

http://kohsuke.org/?s=junit+attachment

### CODE REVIEW

File: src/test_case.js [hunk 1]
```diff
@@ -91,8 +97,13 @@ TestCase.prototype.build = function (parentElement) {
   if (this._standardOutput) {
     testCaseElement.ele('system-out').cdata(this._standardOutput);
   }
+  var systemError = undefined;
   if (this._standardError) {
-    testCaseElement.ele('system-err').cdata(this._standardError);
+    systemError = testCaseElement.ele('system-err').cdata(this._standardError);
+
+    if(this._errorAttachment) {
+      systemError.txt(`[[ATTACHMENT|${this._errorAttachment}]]`);
```
Comment (davidparsson) [inline src/test_case.js]: Ok, I've fixed the original error on `master`, by pinning a development dependency, but the backticks now fails the build on old versions (`0.10`, `0.12`). While I like the syntax, I think I would like to keep this compatible with old versions as well. Does that make sense?

File: spec/e2e_spec.coffee [hunk 2]
```diff
@@ -166,6 +166,20 @@ describe 'JUnit Report builder', ->
...
+    builder.testSuite().testCase()
+      .standardError("This was written to stderr!")
+      .errorAttachment("absolute/path/to/attachment")
+
+    expect(builder.build()).toBe reportWith(
+      '  <testsuite tests="1" failures="0" errors="0" skipped="0">\n' +
+      '    <testcase>\n' +
+      '      <system-err>\n' +
+      '        <![CDATA[This was written to stderr!]]>\n' +
+      '        [[ATTACHMENT|absolute/path/to/attachment]]\n' +
```
Comment (davidparsson) [inline spec/e2e_spec.coffee]: I'm not able to verify that this actually works as expected on Jenkins. Have you tried it?

**Comment by davidparsson** [comment]
If you're able to make the build pass now I'd be more than happy to merge this. Thanks for your contribution!

File: spec/e2e_spec.coffee (hunk 2 above)
Comment (anto-wahanda) [reply spec/e2e_spec.coffee]: The truth is that I haven't tried it. I have implemented the spec by the book. I could find a way to test it on Jenkins but it'd take me some time!

Comment (davidparsson) [reply spec/e2e_spec.coffee]: That would be great! I wouldn't feel that good if I released something that isn't working, and I don't really have the time to set up an environment to try this out myself.

**Comment by davidparsson** [comment]
Now the builds pass, great! Now all I need is a confirmation that it works as intended, and I'll merge it. Please let me know once you've verified that.

**Comment by anto-wahanda** [comment]
Hey @davidparsson. Yes, I've managed to verify it. Behaviour of the junit attachment plugin is even nicer when adding attachment to test cases, as they appear on the test case report page. See images of the xml test report I've used and of the report in Jenkins.

junit-attachment
report

**Comment by davidparsson** [comment]
Nice, thanks! I'll get a new release out as soon as possible.

**Comment by davidparsson** [comment]
Released in `junit-report-builder@1.3.0`.

**Pull request merged**